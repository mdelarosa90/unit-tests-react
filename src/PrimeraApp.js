import React from 'react'
import PropTypes from 'prop-types';

const PrimeraApp = ({saludo, subtitulo }) => {
    // const saludo = 'Hola mundo';
    return (
        <React.Fragment>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <h1>Hola Mundo</h1>
            <h2>{saludo}</h2>
            <p>{subtitulo}</p>
        </React.Fragment>
    )
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subt√≠tulo'
}

export default PrimeraApp
