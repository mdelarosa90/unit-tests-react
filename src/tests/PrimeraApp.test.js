import React from 'react';
import toJson from 'enzyme-to-json';
import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';


describe('Pruebas en <PrimeraApp />', () => {
    // test('debe mostrar mensaje, "Hola Soy Goku"', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const {getByText} = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // });
    test('Debe mostrar <PrimeraAp /> correctamente ', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Soy un Sayayin'
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    })
});