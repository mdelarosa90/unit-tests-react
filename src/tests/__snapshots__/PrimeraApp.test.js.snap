// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraAp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <h2>
    Hola, soy Goku
  </h2>
  <p>
    Soy un subt√≠tulo
  </p>
</Fragment>
`;
